lib/matrix_client/poller.ex

```diff
@@ -20,6 +20,16 @@ defmodule M51.MatrixClient.Poller do
   """
   use Task, restart: :permanent
 
+  require Logger
+
+  # Poller reconnection logic:
+  #  - Initial (re-)connection is always made immediately
+  #  - After that, min delay is added, multiplied by factor on every fail, up to max
+  #  - When connection succeeds, delay is reset
+  @connect_delay_min 30
+  @connect_delay_max 700
+  @connect_delay_factor 1.6
```

Comment:
[`Process.sleep`](https://hexdocs.pm/elixir/1.12/Process.html#sleep/1) works with milliseconds, but it looks like you chose these values as if they were seconds.

30 seconds seems a bit large, though. What about starting with 1000ms, up to 60000ms?

Reply:
Oh, my bad, indeed completely forgot about these s/ms units, will fix.

Yeah, min=1s might be more reasonable for folks who expect to do a lot of chatting, and esp. given "first reconnect might likely be ECONNREFUSED or such".

max=60s seems reasonable if it's to reconnect to server after quick derp, and I guess not too bad for more long-term retries, will adjust to those.
```
# d, d_total, d_min, d_max, d_factor = 0, 0, 1, 60, 1.6
# while d < d_max: d_total += d; print(f'{d:5,.1f} [total {d_total:5,.1f}]'); d = d*1.6 if d else d_min
  0.0 [total   0.0]
  1.0 [total   1.0]
  1.6 [total   2.6]
  2.6 [total   5.2]
  4.1 [total   9.3]
  6.6 [total  15.8]
 10.5 [total  26.3]
 16.8 [total  43.1]
 26.8 [total  69.9]
 42.9 [total 112.9]
```
Though I'll probably adjust it to something like 10min+ locally, to avoid unnecessary log spam, and it doesn't seem to be useful to reconnect that fast for my low-traffic chatting purposes.